app/src/main/java/com/google/android/sambadocumentsprovider/mount/MountServerActivity.java

```diff
@@ -177,14 +215,57 @@ public void onSaveInstanceState(Bundle outState) {
 
     outState.putString(SHARE_PATH_KEY, mSharePathEditText.getText().toString());
     final boolean needsPassword = mNeedPasswordCheckbox.isChecked();
+
     outState.putBoolean(NEEDS_PASSWORD_KEY, needsPassword);
-    if (needsPassword) {
+    outState.putBoolean(AUTH_LAUNCH_KEY, isLaunchedForAuth);
+
+    if (needsPassword || isLaunchedForAuth) {
       outState.putString(DOMAIN_KEY, mDomainEditText.getText().toString());
       outState.putString(USERNAME_KEY, mUsernameEditText.getText().toString());
       outState.putString(PASSWORD_KEY, mPasswordEditText.getText().toString());
     }
   }
 
+  private void prepareForAuth(Intent authIntent) {
```

Comment:
Hmm... Now I feel it has too many if blocks to introduce auth specific logic. We can still have a separate activity for auth, but use the same XML.

TBH, if you would like to use different XML as well, consider using [layout inclusion](https://developer.android.com/training/improving-layouts/reusing-layouts.html) to include common parts.